components/OnboardingChecklist.tsx

```diff
@@ -0,0 +1,303 @@
+import { ReactNode, useCallback, useEffect, useRef, useState } from "react"
+import { ChevronRightIcon } from "@heroicons/react/24/solid"
+import Link from "next/link"
+import clsx from "clsx"
+import { usePredictForm } from "./predict-form/PredictProvider"
+import { api } from "../lib/web/trpc"
+import { useDebounce } from "use-debounce"
+import { QuestionType } from "@prisma/client"
+
+type QuestionCategory = "personal" | "projects" | "shared"
+
+const DEBOUNCE_INTERVAL = 1000
+
+interface GoalSuggestionsProps {
+  category: QuestionCategory
+}
+
+const personalGoals = [
+  "Will I complete my to-do list today?",
+  "Will I be happy I went to <event x>?",
+  "Will I spend <n mins> doing <habit y> today?",
+]
+
+// TODO come up with better examples for these two groups
+const projects = [
+  "Complete the website redesign",
+  "Launch the new marketing campaign",
+  "Develop a new feature for the app",
+]
+
+const shared = [
+  "Organize a surprise party",
+  "Plan a weekend getaway",
+  "Create a photo album",
+]
+
+function GoalSuggestions({ category }: GoalSuggestionsProps) {
+  const [suggestions, setSuggestions] = useState<string[]>([])
+
+  const { setValue } = usePredictForm()
+
+  useEffect(() => {
+    if (category === "personal") {
+      setSuggestions(personalGoals)
+    } else if (category === "projects") {
+      setSuggestions(projects)
+    } else if (category === "shared") {
+      setSuggestions(shared)
+    } else {
+      setSuggestions([])
+    }
+  }, [category])
+
+  // TODO flag decision about the color
+  return (
+    <div className="w-full rounded-b-md flex flex-col items-start gap-2 z-10">
+      {suggestions.map((suggestion) => (
+        <button
+          key={suggestion}
+          className="btn justify-start text-neutral-500 font-medium leading-normal w-full text-neutral-800 border-2 border-neutral-300"
```

Comment:
I'm open to suggestions on the background colour + border of these:
![Screenshot 2024-09-12 at 12 42 47](https://github.com/user-attachments/assets/abb689a4-e116-4b1a-9969-b5086c4e71bc)